crates/miden-objects/src/note/recipient.rs

```diff
@@ -57,9 +57,18 @@ impl NoteRecipient {
         self.digest
     }
 
-    /// Returns the recipient encoded as [Felt]s.
-    pub fn to_elements(&self) -> Vec<Felt> {
-        let mut result = Vec::with_capacity(12);
+    /// Returns the recipient formatted to be used with the advice map.
+    ///
+    /// The format is `inputs_length || INPUTS_COMMITMENT || SCRIPT_ROOT || SERIAL_NUMBER`
+    ///
+    /// Where:
+    /// - inputs_length is the length of the note inputs
+    /// - INPUTS_COMMITMENT is the commitment of the note inputs
+    /// - SCRIPT_ROOT is the commitment of the note script (i.e., the script's MAST root)
+    /// - SERIAL_NUMBER is the recipient's serial number
+    pub fn format_for_advice(&self) -> Vec<Felt> {
+        let mut result = Vec::with_capacity(13);
+        result.push(self.inputs.num_values().into());
```

Comment:
I think this is fine for now, but ideally we'd move towards a setup where a key in the advice map corresponds to the hash of the underlying data. It's not clear to me what's the best way to do this - so, let's just create an issue for this.